src/cic/model.clj

```diff
@@ -47,16 +49,32 @@
       (+ median (* (- upper median) (/ normal 1.96)))
       (- median (* (- median lower) (/ normal -1.96))))))
 
+(defn clamp
+  [lower x upper]
+  (max (min x upper) lower))
+
 (defn duration-model
   "Given an admitted date and age of a child in care,
   returns an expected duration in days"
   [coefs]
-  (fn [age seed]
-    (let [empirical (get coefs (max 0 (min age 17)))
-          [r1 r2] (rand/split seed)
-          quantile (int (p/sample-1 (d/uniform {:a 1 :b 101}) r1))
-          [lower median upper] (get empirical quantile)]
-      (sample-ci lower median upper r2))))
+  (fn duration-model*
+    ([birthday beginning seed]
+     (duration-model* birthday beginning 0 seed))
+    ([birthday beginning min-value seed]
+     (let [age (time/year-interval birthday beginning)
+           empirical (get coefs (max 0 (min age 17)))
+           n (transduce (take-while (fn [[_ m _]] (<= m min-value))) k/count empirical)
+           [r1 r2] (rand/split seed)
+           max-value (dec (time/day-interval birthday (time/years-after birthday 18)))]
+       (if (> n 100)
+         (int (p/sample-1 (d/uniform {:a min-value :b max-value}) r1))
+         (loop [r1 r1 iter 1]
```

Comment:
this feels like it wants to be a `(->> lazy-seq-of-durations (take 100) (some #(when (< min-value sample) sample)))` but I'd have to work through it a bit more (or have I misunderstood what is going on here?